### DISCUSSION TITLE
Accessing request location and device Information from server actions

### DISCUSSION THREAD

**Discussion by InsightfulFuture** [vercel/next.js#65219] [Help]
### Summary

NextRequest provides easy access to geo items like these if you're hosted on Vercel:
request.geo.city
request.geo.country
request.geo.region

userAgent offers access to:
browser.name
device.model
etc.

I'm using server actions and love them, but is it true that request related data elements like those noted above are not accessible from server actions?

I can get the request ip in a server action like this:
headers().get('x-forwarded-for');

but is it possible to access the additional items noted in these docs? This would be very helpful:
https://nextjs.org/docs/app/api-reference/functions/next-request
https://nextjs.org/docs/app/api-reference/functions/userAgent

If these are not accessible directly within server actions... is there a workaround (middleware, etc.)?

**Comment by TxHawks** [comment]
I'm also in need of geolocation inside a server action, and not having access to the request object seems to mean it's not possible
